### PR TITLE
audiowaveform: add initial recipe for 1.10.1

### LINKED ISSUE #9371: [request] audiowaveform/1.5.1
### Package Details
  * Package Name/Version: **audiowaveform/1.5.1**
  * Website: **https://github.com/bbc/audiowaveform**
  * Source code: **https://github.com/bbc/audiowaveform**

### Description Of The Library / Tool

audiowaveform is a C++ command-line application that generates waveform data from either MP3, WAV, FLAC, Ogg Vorbis, or Opus format audio files. Waveform data can be used to produce a visual rendering of the audio, similar in appearance to audio editing applications.

### PR DESCRIPTION
**Pull request by Nekto89**
Specify library name and version:  **audiowaveform/1.10.1**

Resolves #9371

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Nekto89** [comment]
TODO: check after #18987 is merged

File: recipes/audiowaveform/all/conanfile.py
Comment (valgur) [inline recipes/audiowaveform/all/conanfile.py:1]: Needs a `package_id()` with `del self.settings.compiler`.

File: recipes/audiowaveform/all/patches/1.9.1-disable_documentation.patch
Comment (valgur) [inline recipes/audiowaveform/all/patches/1.9.1-disable_documentation.patch:1]: Just delete them after `cmake.install()`. Easier to maintain.

File: recipes/audiowaveform/all/test_v1_package/conanfile.py
Comment (valgur) [inline recipes/audiowaveform/all/test_v1_package/conanfile.py:1]: `test_v1_package` should no longer be included in new recipes.

**Review by valgur — CHANGES_REQUESTED** [review]
Needs a `package_id()`, otherwise LGTM.

I'm not sure if CCI is the best place for this package, though, since it's not really a library or a build tool.

File: recipes/audiowaveform/all/patches/1.9.1-disable_documentation.patch
Comment (Nekto89) [reply recipes/audiowaveform/all/patches/1.9.1-disable_documentation.patch:1]: documentation requires "gzip" in PATH. In my opinion it's easier to just disable it.

Comment (valgur) [reply recipes/audiowaveform/all/patches/1.9.1-disable_documentation.patch:1]: Ah, ok. Makes sense.

**Review by valgur — APPROVED** [review]

**Review by perseoGI — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**
